src/main/scala/org/ergoplatform/local/ErgoStatsCollector.scala

```diff
@@ -17,29 +20,46 @@ import scorex.core.network.peer.PeerManager.ReceivableMessages.GetConnectedPeers
 import scorex.core.utils.{NetworkTimeProvider, ScorexLogging}
 
 import scala.concurrent.duration._
+import scala.language.postfixOps
 
 /**
   * Class that subscribes to NodeViewHolderEvents and collects them to provide fast response to API requests.
   */
-class ErgoStatsCollector(viewHolderRef: ActorRef,
+class ErgoStatsCollector(readersHolder: ActorRef,
                          peerManager: ActorRef,
                          settings: ErgoSettings,
                          timeProvider: NetworkTimeProvider)
   extends Actor with ScorexLogging {
 
+  implicit val ec = context.system.dispatcher
+  implicit val timeout = Timeout(5 seconds)
+
   override def preStart(): Unit = {
     context.system.eventStream.subscribe(self, classOf[ChangedHistory[_]])
     context.system.eventStream.subscribe(self, classOf[ChangedMempool[_]])
     context.system.eventStream.subscribe(self, classOf[SemanticallySuccessfulModifier[_]])
-    context.system.scheduler.schedule(10.second, 10.second)(peerManager ! GetConnectedPeers)(context.system.dispatcher)
+    context.system.scheduler.schedule(10.second, 10.second)(peerManager ! GetConnectedPeers)
+
+    (readersHolder ? GetReaders).mapTo[Readers].pipeTo(self)
```

Comment:
I think readersHolder ! GetReaders should be enough, no need to wait the response